### PR TITLE
correctly handles a date object when a dimension is casted to time column

### PR DESCRIPTION
**Pull request by chrismclaughlin55**

### CODE REVIEW

File: test/simulate/simulateDruid.mocha.js [hunk 1]
```diff
@@ -3836,6 +3836,42 @@ describe('simulate Druid', () => {
     });
   });
 
+  it('TEST TEST yaa', () => {
```
Comment (davidagee) [inline test/simulate/simulateDruid.mocha.js:3839]: we should probably have a more descriptive name for the test

File: src/external/utils/druidExpressionBuilder.ts [hunk 2]
```diff
@@ -143,6 +143,10 @@ export class DruidExpressionBuilder {
           case 'boolean':
             return String(Number(literalValue));
 
+          case 'object':
+            if ( literalValue instanceof Date) return DruidExpressionBuilder.escapeLiteral(literalValue);
```
Comment (wylieallen-i) [inline src/external/utils/druidExpressionBuilder.ts:147]:
```suggestion
            if (literalValue instanceof Date) return DruidExpressionBuilder.escapeLiteral(literalValue);
```
Plywood doesn't have Prettier I guess?

File: test/simulate/simulateDruid.mocha.js [hunk 3]
```diff
@@ -3836,6 +3836,42 @@ describe('simulate Druid', () => {
...
+      allowSelectQueries: true,
+      filter: $('time').overlap({
+        start: new Date('2015-03-12T00:00:00Z'),
+        end: new Date('2015-03-19T00:00:00Z'),
+      }),
+      context: {
+        priority: -1,
+        queryId: 'test',
+      },
+    });
```
Comment (wylieallen-i) [inline test/simulate/simulateDruid.mocha.js:3854]: Does this do anything? I don't see `ds` referenced anywhere in the test

**Review by wylieallen-i — CHANGES_REQUESTED** [review]

**Review by wylieallen-i — APPROVED** [review]
:rocket:

**Pull request merged**